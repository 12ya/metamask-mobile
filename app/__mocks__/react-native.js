import { Linking, DeviceEventEmitter } from 'react-native';

Linking.addEventListener = jest.fn();
Linking.removeEventListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeEventListener
  if (typeof callback === 'function') {
    callback();
  }
});
Linking.openURL = jest.fn();
Linking.canOpenURL = jest.fn();
Linking.getInitialURL = jest.fn();

DeviceEventEmitter.addListener = jest.fn();
DeviceEventEmitter.removeListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeListener
  if (typeof callback === 'function') {
    callback();
  }
});

export default { Linking, DeviceEventEmitter };
