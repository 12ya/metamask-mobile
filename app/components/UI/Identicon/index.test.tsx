import React from 'react';
import { render } from '@testing-library/react-native';
import Identicon from './';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

describe('Identicon', () => {
  const mockStore = configureMockStore();
  it('should render correctly when useBlockieIcon is true', () => {
    const initialState = {
      settings: { useBlockieIcon: true },
    };
    const store = mockStore(initialState);

    const { toJSON } = render(
      <Provider store={store}>
        <Identicon />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('should render correctly when useBlockieIcon is false', () => {
    const initialState = {
      settings: { useBlockieIcon: false },
    };
    const store = mockStore(initialState);

    const { toJSON } = render(
      <Provider store={store}>
        <Identicon />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
