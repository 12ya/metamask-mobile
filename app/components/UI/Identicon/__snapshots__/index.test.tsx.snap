// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Identicon should render correctly when provided address found in tokenList and iconUrl is available 1`] = `
<Image
  source={
    {
      "uri": "https://example.com/icon.png",
    }
  }
  style={
    [
      {
        "borderRadius": 23,
        "height": 46,
        "width": 46,
      },
      undefined,
    ]
  }
/>
`;

exports[`Identicon should render correctly when useBlockieIcon is false 1`] = `null`;

exports[`Identicon should render correctly when useBlockieIcon is true 1`] = `null`;
